pallets/balances/src/lib.rs

```diff
@@ -462,6 +462,22 @@ decl_module! {
             let dest = T::Lookup::lookup(dest)?;
             Self::transfer_core(&source, &dest, value, None, ExistenceRequirement::AllowDeath)?;
         }
+
+        /// Burns the given amount of tokens from the caller's free balance.
+        #[weight = SimpleDispatchInfo::FixedNormal(200_000)]
+        pub fn burn_free_balance(origin, amount: T::Balance) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            let total_issuance = Self::total_issuance();
+            let amount = cmp::min(total_issuance, cmp::min(amount, Self::account(&who).free));
+            // Burn total issuance.
+            <TotalIssuance<T>>::put(total_issuance - amount);
```

Comment:
Isn't it better to use the `Currency::withdraw()` to get the negative imbalance and to balanced that you can use `Currency::burn()` which will give you a positive imbalance. Where +ve imbalance should be equal to -ve imbalance

Reply:
Wouldn't that however do the same thing but in more steps?

Reply:
@satyamakgec Substrate doesn't offer a `WithdrawReason` for burning the balance. This is a technical issue with the proposal to use `withdraw`. Unless you wish to redefine `WithdrawReason` on our side with all the methods that reference it, I'd prefer the simpler solution as currently implemented.

Reply:
I was more concerned related to the checks that are present in those functions like `ensure_can_withdraw()` etc. So we don't miss anything

Reply:
then it will be better to copy those checks here if we are not able to call them directly because of some reasons.

Reply:
We can call them but we need to decide how to circumvent the absence of a "burn" bit flag. Substrate has
```
	flags WithdrawReason {
		/// In order to pay for (system) transaction costs.
		TransactionPayment = 0b00000001,
		/// In order to transfer ownership.
		Transfer = 0b00000010,
		/// In order to reserve some funds for a later return or repatriation.
		Reserve = 0b00000100,
		/// In order to pay some other (higher-level) fees.
		Fee = 0b00001000,
		/// In order to tip a validator for transaction inclusion.
		Tip = 0b00010000,
	}
```

Reply:
I'll try `WithdrawReasons::none()`. It typechecks but needs to be tested.